firedrake/slate/slate.py

```diff
@@ -821,6 +825,24 @@ def arguments(self):
         return self._args
 
 
+def space_equivalence(A, B):
+    """Checks that two function spaces are equivalent.
+
+    :arg A: A function space.
+    :arg B: Another function space.
+
+    Returns `True` if they have matching meshes, elements, and rank. Otherwise,
+    `False` is returned.
+    """
+
+    equiv = False
+    assert A.mesh() == B.mesh(), "Mismatching meshes!"
+    if A.rank == B.rank and A.ufl_element() == B.ufl_element():
+        equiv = True
+
+    return equiv
```

Comment:
You can do beta-reduction:

`return A.mesh() == B.mesh() and A.ufl_element() == B.ufl_element() and A.rank == B.rank`.